### PR TITLE
Inner-Loop Setup with DevSpace

### PR DESCRIPTION
**Pull request by sudivate**
1. Inner Loop Setup with Devspace for Azure Vote Front-end and back-end app 
2. Scripts to generate manifest required for working in Inner-Loop
2. Document that describes how to use Devspace for Inner-Loop

### CODE REVIEW

File: azure-vote/manifests/azure-vote-front/kustomize/base/kustomization.yaml [hunk 1]
```diff
@@ -1,12 +1,11 @@
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 
-namespace: $TARGET_NAMESPACE
+namespace: voteapp
```
Comment (eedorenko) [inline azure-vote/manifests/azure-vote-front/kustomize/base/kustomization.yaml:4]: Is it intentionally? The namespace is always voteapp? TARGET_NAMESPACE is not used?

**Review by eedorenko — APPROVED** [review]

File: utils/generate-manifests-dev.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+# Generates K8s manifests from Helm + Kustomize templates
+# Uses env variables to substitute values 
+# Requires to be installed:  
+#   - helm 
+#   - kubectl
+#   - envsubst (https://command-not-found.com/envsubst)
+# 
+
+# For the Azure-Vote sample the following env varuables are supposed to be initialized:
```
[the remark this answers is not in the document]
Comment (sudivate) [reply utils/generate-manifests-dev.sh]: Added .env and updated instructions

File: azure-vote/manifests/azure-vote-front/kustomize/base/kustomization.yaml (hunk 1 above)
Comment (sudivate) [reply azure-vote/manifests/azure-vote-front/kustomize/base/kustomization.yaml]: good catch. It should be  $TARGET_NAMESPACE

File: docs/inner-loop.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+### Inner-Loop Flow 
+
+
+1. Run `devspace use context` and select right kubernetes cluster (local or remote) 
+1. Create a dedicate namespace by running `devspace use namespace <namespace_name>`.
+1. Run `devspace run gen-manifest` to generate manifest Kustomization required to run 'Azure Vote App'.
+1. Post making changes to application code run `devspace dev` to instantly run Azure front-end container along with its dependencies in target cluster.
+1. Post running `devspace dev`, you can access Azure front-end through the port forwarding configuration specified in devspace.yaml e.g `http://localhost:8085/`
+1. Running `devspace dev` also syncs app files into containers and you can continue making changes to app code without rebuilding image.
+1. After the changes are finalized and tested push a new image to Azure Container registry by running `devspace build`
```
[the remark this answers is not in the document]
Comment (sudivate) [reply docs/inner-loop.md]: Added an implicit hook to generate manifest  on `devspace dev` . However you can still run `devspace run gen-manifest` to explicitly generate manifest

**Pull request merged**